sunpy/timeseries/timeseries_factory.py

```diff
@@ -287,7 +289,11 @@ def _parse_args(self, *args, **kwargs):
                                 # to preserve multi-line comments
                                 if isinstance(args[i+1], astropy.io.fits.header.Header):
                                     args[i+1] = MetaDict(sunpy.io.header.FileHeader(args[i+1]))
-                                meta.update(args[i+1])
+                                if isinstance(args[i+1], sunpy.timeseries.TimeSeriesMetaData):
+                                    for j in args[i+1].metas:
+                                        meta.update(j)
```

Comment:
I don't follow why you added this? Is there something we have to do with `TimeSeriesMetaData` to get it to work?

Reply:
Simply passing the ``TimeSeriesMetaData`` object was causing an error while the ``meta`` was being updated. As ``TimeSeriesMetaData`` object has a list of ``MetaDict`` objects, I simply updated the the meta with it. 
An alternative was to write separate conditions for the ``TimeSeriesMetaData`` object everywhere.

Reply:
Fair enough but could I get a test for this bit of code?

Reply:
Is the test I added satisfactory?

Reply:
Works for me.